cpp/serve/request.h

```diff
@@ -0,0 +1,222 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file request.h
+ * \brief Implementation of llm chat.
+ */
+#ifndef MLC_LLM_SERVE_REQUEST_H_
+#define MLC_LLM_SERVE_REQUEST_H_
+
+#include <tvm/node/reflection.h>
+#include <tvm/node/structural_equal.h>
+#include <tvm/node/structural_hash.h>
+#include <tvm/runtime/container/array.h>
+#include <tvm/runtime/container/optional.h>
+#include <tvm/runtime/container/shape_tuple.h>
+#include <tvm/runtime/container/string.h>
+#include <tvm/runtime/object.h>
+
+#include "config.h"
+#include "input.h"
+
+namespace mlc {
+namespace llm {
+namespace serve {
+
+using namespace tvm::runtime;
+
+/****************** Request ******************/
+
+/*!
+ * \brief The user submitted text-generation request, which contains
+ * a list of multi-modal inputs and a set of sampling configuration
+ * parameters.
+ */
+class RequestNode : public Object {
+ public:
+  Array<Input> inputs;
+  /*! \brief JSON string containing temperature, top_p, repetition_penalty, max_gen_len, etc. */
+  SampleConfig sampling_params;
```

Comment:
cross check name `sample_params`: let us make it consistent with openAI API

Reply:
OpenAI API and TGI do not have a standalone data class for the config here
* https://platform.openai.com/docs/api-reference/chat/create
* https://huggingface.github.io/text-generation-inference/#/Text%20Generation%20Inference/generate

vLLM uses class class name `SamplingParams` and instance name `sampling_params`.
* https://github.com/vllm-project/vllm/blob/main/examples/llm_engine_example.py#L24-L33

Reply:
Let us update SampleConfig=>SamplingParams then, so we can be consistent